sklearn/tree/_export.py

```diff
@@ -919,6 +921,16 @@ def compute_depth_(
     return max(depths)
 
 
+@validate_params(
+    {
+        "decision_tree": [DecisionTreeClassifier, DecisionTreeRegressor],
+        "feature_names": [list, None],
+        "max_depth": [Interval(Integral, -1, None, closed="left"), None],
```

Comment:
max_depth must be >= 0
```suggestion
        "max_depth": [Interval(Integral, 0, None, closed="left"), None],
```

Reply:
@jeremiedbb thanks for the helpful suggestions. 

However, when I use `"max_depth": [Interval(Integral, 0, None, closed="left"), None]` I seem to be getting errors from `pytest sklearn/tree/test_export.py` from function ` test_export_text_errors()`

Error:
`During handling of the above exception, another exception occurred:

    def test_export_text_errors():
        clf = DecisionTreeClassifier(max_depth=2, random_state=0)
        clf.fit(X, y)
    
        err_msg = "max_depth bust be >= 0, given -1"
>      with pytest.raises(ValueError, match=err_msg):
E       AssertionError: Regex pattern 'max_depth bust be >= 0, given -1' does not match "The 'max_depth' parameter of export_text must be an int in the range [0, inf) or None. Got -1 instead.".

sklearn/tree/tests/test_export.py:355: AssertionError`

Similar issues for decimals and spacing. Since I am a beginner at this, can you please provide suggestions for these? Thanks!